manifests/params.pp

```diff
@@ -4,7 +4,7 @@
 # It sets variables according to platform.
 #
 class puppet_agent::params {
-  if (versioncmp("${::clientversion}", '4.0.0') < 0 and $::puppet_stringify_facts == true) {
+  if (versioncmp("${::clientversion}", '4.0.0') < 0 and str2bool($::puppet_stringify_facts == true)) {
```

Comment:
Did you mean to str2bool the equality check, rather than puppet_stringify_facts?

Reply:
Whoops, let me fix this